File: obiektowe/lista5/zad2/Const.java
import java.util.HashMap;

public class Const extends Expression
{
    private int value;

    public Const(int value)
    {
        this.value=value;
    }

    public int eval()
    {
        return value;
    }

    public int eval(HashMap<String,Integer> vals)
    {
        return this.eval();
    }

    public String toString()
    {
        return Integer.toString(value);
    }
}

File: cpp/zad6/Const.cpp
#include"Expression.hpp"

Pi::Pi()
{
    this->value = M_PI;
}
double Pi::eval()
{
    return this->value;
}
std::string Pi::to_string()
{
    return std::to_string(value);
}

e::e()
{
    this->value = M_E;
}
double e::eval()
{
    return this->value;
}
std::string e::to_string()
{
    return std::to_string(value);
}

Phi::Phi()
{
    this->value = 1.61803398875;
}
double Phi::eval()
{
    return this->value;
}
std::string Phi::to_string()
{
    return std::to_string(value);
}